nuxeo-retention/src/main/java/org/nuxeo/retention/service/RetentionManagerImpl.java

```diff
@@ -185,6 +197,34 @@ protected void checkCanAttachRule(DocumentModel document, RetentionRule rule, Co
         }
     }
 
+    @Override
+    public void fireRetentionEvent(String eventName, String eventInput, boolean audit, CoreSession session) {
+        NuxeoPrincipal principal = session.getPrincipal();
+        if (!principal.isAdministrator() && !principal.isMemberOf(RECORD_MANAGER_GROUP_NAME)) {
+            throw new NuxeoException(
+                    String.format("User: %s is not authorized to fire retention event", principal.getPrincipalId()),
+                    SC_FORBIDDEN);
+        }
+        if (StringUtils.isNotBlank(eventInput) && !EVENT_INPUT_PATTERN.matcher(eventInput).matches()) {
+            throw new IllegalArgumentException("Invalid retention event input: " + eventInput
+                    + "  is not following the expected pattern:" + EVENT_INPUT_REGEX);
```

Comment:
How rude, you're throwing to the user a stacktrace with a regex :rofl:

Reply:
:shrug:

Reply:
Web UI team will probably add an input validation message :joy: